src/include/miopen/tensor.hpp

```diff
@@ -213,8 +247,10 @@ struct TensorDescriptor : miopenTensorDescriptor
     std::vector<std::size_t> strides;
 
     bool packed;
+    int vector_length = 1;
 
-    miopenDataType_t type = miopenFloat;
+    miopenDataType_t type             = miopenFloat;
+    miopenTensorLayout_t tensorLayout = miopenTensorNCHW;
```

Comment:
I disagree with this, on a number two aspects:
 - tensor layout shouldn't be of enum type, (rather it should be dynamically calculated and cached as string). Using a enum type will make it much harder to scale when we are to add a different layout. This has been concluded in https://github.com/ROCmSoftwarePlatform/MIOpen/pull/343
 - tensor layout shouldn't be cached in `tensor` type, but in `problem_description`

With it, I prefer the changes in here/constructor to be reverted such that a client of tensor descriptor doesn't need to use a layout to create a tensor descriptor. If I remember it right, @pfultz2 was against this opinion when I was trying to do (almost) the same change in "ambiguous layout" fix in https://github.com/ROCmSoftwarePlatform/MIOpen/issues/767.

Reply:
@aska-0096 @carlushuang  for awareness

Reply:
@jerryyin 
> tensor layout shouldn't be cached in tensor type, but in problem_description

IIRC the modern convolution's problem description holds tensor descriptors, which, in turn, hold layouts. Is it acceptable?

Reply:
> convolution's problem description holds tensor descriptors, which, in turn, hold layouts

Yes, that sounds like the consistent approach with the infrastructure already exists before this PR.

Reply:
Yea this is wrong. The layout is a property of the strides. If a user passes the layout with strides instead of an enum this will be broken.